ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/editorhook/js/ruleeditorhook.js

```diff
@@ -34,17 +35,21 @@
         if (!formContainerPath) {
             showAlert();
         } else {
-            let ruleEditorUri = '/aem/af/expeditor.html' + getFormContainerPath(editable) + "?fieldPath=" + editable.path + "&fieldId=" + getFieldId(editable);
-            ruleEditorFrame.setAttribute('src', ruleEditorUri);
-            ruleEditorFrame.setAttribute('title', 'AF Rule Editor');
-            ruleEditorFrame.style.display = "block";
-            ruleEditorFrame.style.width = "100%";
-            ruleEditorFrame.style.height = "100%";
-            ruleEditorFrame.style.top = "0";
-            ruleEditorFrame.style.left = "0";
-            ruleEditorFrame.style.position = "fixed";
-            ruleEditorFrame.style.zIndex = "10";
-            document.body.appendChild(ruleEditorFrame);
+            if (typeof window.CQ.FormsCoreComponents.editorhooks.v2.openRuleEditor === 'function') {
+                window.CQ.FormsCoreComponents.editorhooks.v2.openRuleEditor(editable);
+            } else {
+                let ruleEditorUri = '/aem/af/expeditor.html' + getFormContainerPath(editable) + "?fieldPath=" + editable.path + "&fieldId=" + getFieldId(editable);
```

Comment:
This can be done like this,

```
const ruleEditorUri = `/aem/af/expeditor.html${getFormContainerPath(editable)}?fieldPath=${editable.path}&fieldId=${getFieldId(editable)}`;
ruleEditorFrame.setAttribute('src', ruleEditorUri);
ruleEditorFrame.setAttribute('title', 'AF Rule Editor');

const styles = {
    display: "block",
    width: "100%",
    height: "100%",
    top: "0",
    left: "0",
    position: "fixed",
    zIndex: "10"
};

Object.assign(ruleEditorFrame.style, styles);
document.body.appendChild(ruleEditorFrame);
```